File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    count_zero = 0
    count=0
    
    while s != '1' :
        count_one=0
        for i in range(0,len(s)) :
            if s[i] == '1' :
                count_one +=1
        count_zero += len(s) - count_one
        s = format(count_one,'b')
        print(s)
        count+=1
        
            
    answer.append(count)
    answer.append(count_zero)
    return answer

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int count_zero = 0;
        int count_while = 0;
        int[] answer = new int[2];
       
        while (!s.equals("1")){
            count_while ++;
            String del_zero = "";
            
            for(int i =0; i<s.length(); i++){
                if(s.charAt(i) =='1'){
                    del_zero += "1";
                } else{
                    count_zero ++;
                }
            }
            s = Integer.toBinaryString(del_zero.length());
        }
        
        answer[0] = count_while;
        answer[1] = count_zero;

        return answer;
    }

}